crates/e2e/src/subxt_client.rs

```diff
@@ -96,6 +96,27 @@ use subxt::{
 
 pub type Error = crate::error::Error<DispatchError>;
 
+/// Trait for types that can be converted to account bytes.
+///
+/// This allows unified API methods to accept both Keypair and AccountId types.
+/// Similar to the sandbox's `IntoAccountId` trait.
+pub trait ToAccountBytes {
+    fn to_account_bytes(&self) -> [u8; 32];
+}
+
+impl ToAccountBytes for Keypair {
+    fn to_account_bytes(&self) -> [u8; 32] {
+        self.public_key().0
+    }
+}
+
+impl ToAccountBytes for ink::primitives::AccountId {
```

Comment:
Do we really need to introduce a new trait for this?  Please check if one of the traits that `AccountId` already implements is also implented by `Keypair`: https://github.com/use-ink/ink/blob/master/crates/primitives/src/types.rs#L102-L106.